Template_Skill/src/main/java/com/wolfpack/database/DbConnection.java

```diff
@@ -0,0 +1,248 @@
+	 *       connection.
+	 * 
+	 * @return true if the database was successfully connected to. Otherwise,
+	 *         false if there were any errors.
+	 * 
+	 */
+	public boolean getRemoteConnection() {
+
+		// Create a url to be able to connect to the database.
+		String jdbcUrl = "jdbc:postgresql://" + hostName + ":" + port + "/" + dbName + "?user=" + username
+				+ "&password=" + password + "&sslmode=verify-full&sslrootcert=" + localPathToSSL;
+
+		try {
+			Class.forName("org.postgresql.Driver");
+			dbConnection = DriverManager.getConnection(jdbcUrl);
+
+		// Error handling
+		} catch (ClassNotFoundException e1) {
+			System.out.println("Driver needed for connecting to the postgresql database.");
+			return false;
+		} catch (SQLException e2) {
+			System.out.println("Unable to establish a connection to the database.");
+			return false;
+		}
+
+		return true;
+
+	}
+
+	/**
+	 * @return true if the database has been connected to, otherwise false if
+	 *         there is not a current connection.
+	 */
+	public static boolean isConnected() {
+		return connected;
+	}
+
+	/**
+	 * The runQuery() method will attempt to run the SQL query passed to it and
+	 * then return a Map object that uses the column names as they keys and
+	 * vectors of what was in the columns as the values for the map. All of the
+	 * values get stored as strings in the vectors.
+	 * 
+	 * As an example, if your query returned this table: movieId | name
+	 * ---------------------- 123 | Tron 456 | Harry Potter
+	 * 
+	 * The resulting map would look like this: Key = "movieId" : Value = {"123",
+	 * "456"} Key = "name" : Value = {"Tron", "Harry Potter"}
+	 * 
+	 * In the vectors, the items from a row will all share the same index. So in
+	 * the example, the first row's information could be grabbed by:
+	 * vectorName.get("movieId")[0]; vectorName.get("name")[0];
+	 * 
+	 * @return Map<String, Vector<String>> object representing the query's
+	 *         results if everything went successfully. Otherwise, null.
+	 */
+	public Map<String, Vector<String>> runQuery(String query) {
+		Map<String, Vector<String>> resultMap = new HashMap<String, Vector<String>>();
+		try {
+			Statement st = dbConnection.createStatement();
+			ResultSet rawResults = st.executeQuery(query);
+			ResultSetMetaData rsmd = rawResults.getMetaData();
+
+			// Set up all of the keys (column names) with empty vectors as the
+			// values
+			for (int i = 1; i <= rsmd.getColumnCount(); i++) {
+				String name = rsmd.getColumnName(i);
+				resultMap.put(name, new Vector<String>());
+			}
+
+			// Push the values into the vectors
+			while (rawResults.next()) {
+				for (int i = 1; i <= rsmd.getColumnCount(); i++) {
+					resultMap.get(rsmd.getColumnName(i)).addElement(rawResults.getString(i));
+				}
+			}
+		}
+		// Error checking
+		catch (SQLException e) {
+			System.out.println("Problem with received query. (Perhaps it is an invalid query?)");
```

Comment:
Can you print the message from the SQLException here, too? That would probably be helpful for debugging.